lib/features/push_notification/presentation/controller/web_socket_controller.dart

```diff
@@ -63,6 +64,23 @@ class WebSocketController extends PushBaseController {
     super.initialize(accountId: accountId, session: session);
 
     _connectWebSocket(accountId, session);
+    if (!PlatformInfo.isWeb) {
+      _listenToAppLifeCycle(accountId, session);
+    }
+  }
+
+  void _listenToAppLifeCycle(AccountId? accountId, Session? session) {
+    _appLifecycleListener ??= AppLifecycleListener(
```

Comment:
It's not possible. This listener react according to the app's state. So
- The app is moved to background/terminated, `AppLifecycleListener` triggers
-  `AppLifecycleListener` close web socket connection by `_closeWebSocket`
- If we dispose the `AppLifecycleListener` here, next time the app is reopened, `AppLifecycleState.resumed` of `AppLifecycleListener` will not get triggered, hence web socket will not reconnect.

Reply:
So we have to dispose it before WebSocketController is destroyed, to avoid memory leak.

Reply:
There is no memory leak. See profile demo below

Reply:
In the memory leak case you demoed, only the WebSocketController is not destroyed because it is in the background state. Let's test the case where the WebSocketController is destroyed.

Reply:
listener disposed